packages/@react-spectrum/contextualhelp/src/ContextualHelp.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ActionButton} from '@react-spectrum/button';
+import {Content, Footer} from '@react-spectrum/view';
+import {Dialog, DialogTrigger} from '@react-spectrum/dialog';
+import {Heading} from '@react-spectrum/text';
+import HelpOutline from '@spectrum-icons/workflow/HelpOutline';
+import helpStyles from './contextualhelp.css';
+import InfoOutline from '@spectrum-icons/workflow/InfoOutline';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {Link} from '@react-spectrum/link';
+import React, {ReactElement} from 'react';
+import {SlotProvider} from '@react-spectrum/utils';
+import {SpectrumContextualHelpProps} from '@react-types/contextualhelp';
+import {useMessageFormatter} from '@react-aria/i18n';
+
+function ContextualHelp(props: SpectrumContextualHelpProps) {
+  let {
+    variant = 'help',
+    placement = 'bottom end',
+    children,
+    ...otherProps} = props;
+
+  let formatMessage = useMessageFormatter(intlMessages);
+
+  let icon = variant === 'info' ? <InfoOutline /> : <HelpOutline />;
+  let link: ReactElement | null = null;
+
+  let dialogSlots: ReactElement[] = [];
+  React.Children.forEach(children, child => {
+    if (React.isValidElement(child)) {
+      if (child.type === Link) {
```

Comment:
pull out `Link` child so it can be wrapped in a `Footer` slot first.

Reply:
This actually defeats the purpose of slots. I'd prefer to have our own layout and completely null out Dialog's layout and context before we did something like this.
Though if we're going to null it all out, it may make sense to just recreate as opposed to potentially nulling some things but forgetting other things.

Another good reason to keep Footer would be to future support the multi link case you mentioned

Reply:
OK, so if we decide to simply go with the existing `Dialog` slots do we force the Footer to only contain links or can it contain anything?

```
    <ContextualHelp>
      <Heading>Help title</Heading>
      <Content>Help text</Content>
      <Footer>
        <Link>Help link</Link>
        <Button>Button</Link>
      </Footer>
    </ContextualHelp>
```

Reply:
technically it could have anything, much like many of our components, we rely on Spectrum Design guidelines to help mitigate putting just anything in there
there are certain elements, that if you put in here, will be automatically placed and styled as Spectrum has designed

Reply:
@snowystinger I have updated the PR to have `ContextualHelp` just use the existing `Dialog` slots. This simplifies the implementation and just applies spectrum design styling on top of existing dialog pattern.

Reply:
thanks! I'm getting Danni and Devon to check it out as well